tests/test_hyperparameter_retrieval.py

```diff
@@ -0,0 +1,40 @@
+"""
+unit test for hyperparameter parsing
+"""
+import pytest
+from domainlab.arg_parser import mk_parser_main
+from domainlab.utils.hyperparameter_retrieval import get_gamma_reg
+
+def test_store_dict_key_pair_single_value():
+    """Test to parse a single gamma_reg parameter"""
+    parser = mk_parser_main()
+    args = parser.parse_args(['--gamma_reg', '0.5'])
+    assert args.gamma_reg == 0.5
+
+def test_store_dict_key_pair_dict_value():
+    """Test to parse a dict for the gamma_reg"""
+    parser = mk_parser_main()
+    args = parser.parse_args(['--gamma_reg', 'dann=1.0,jigen=2.0'])
+    assert args.gamma_reg == {'dann': 1.0, 'jigen': 2.0}
+
+def test_get_gamma_reg_single_value():
+    """Test to retrieve a single gamma_reg parameter which is applied to all objects"""
+    parser = mk_parser_main()
+    args = parser.parse_args(['--gamma_reg', '0.5'])
+    assert get_gamma_reg(args, 'dann') == 0.5
+
+def test_get_gamma_reg_dict_value():
+    """Test to retrieve a dict of gamma_reg parameters for different objects"""
+    parser = mk_parser_main()
+    args = parser.parse_args(['--gamma_reg', 'default=5.0,dann=1.0,jigen=2.0'])
+    assert get_gamma_reg(args, 'dann') == 1.0
+    assert get_gamma_reg(args, 'jigen') == 2.0
+    assert get_gamma_reg(args, 'nonexistent') == 5.0
```

Comment:
what does 'nonexistent' mean here?

Reply:
what should be the behavior here for "nonexistent"?  @MatteoWohlrapp

Reply:
@MatteoWohlrapp , shouldn't that raise an error? since nonexistence is not implmented as a model or trainer?  or any other design logic here?

Reply:
I see, actually nonexsistence can represent anything that we have not implmented, so no issue here.

Reply:
added a comment in [0d62919](https://github.com/marrlab/DomainLab/pull/806/commits/0d629194fa559e63abbc6c1eadbd0685d87f1c2a)[0d62919](https://github.com/marrlab/DomainLab/pull/806/commits/0d629194fa559e63abbc6c1eadbd0685d87f1c2a)